fiftyone/utils/eval/coco.py

```diff
@@ -5,326 +5,180 @@
 | `voxel51.com <https://voxel51.com/>`_
 |
 """
-# pragma pylint: disable=redefined-builtin
-# pragma pylint: disable=unused-wildcard-import
-# pragma pylint: disable=wildcard-import
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
-from builtins import *
-
-# pragma pylint: enable=redefined-builtin
-# pragma pylint: enable=unused-wildcard-import
-# pragma pylint: enable=wildcard-import
-
-import logging
+from collections import defaultdict
 
 import numpy as np
 
-import fiftyone.core.media as fom
-import fiftyone.core.utils as fou
-
-
-logger = logging.getLogger(__name__)
+import eta.core.utils as etau
 
+from .detection import DetectionEvaluationConfig, DetectionEvaluationMethod
 
-def evaluate_detections(
-    samples,
-    pred_field,
-    gt_field="ground_truth",
-    iou=0.75,
-    classwise=True,
-    save_sample_fields=True,
-):
-    """Evaluates the predicted detections in the given samples with respect to
-    the specified ground truth detections using the specified Intersection over
-    Union (IoU) threshold to determine matches.
 
-    This method uses COCO-style evaluation. In particular, this means that if a
-    :class:`fiftyone.core.labels.Detection` in the ground truth field has a
-    boolean attribute called ``iscrowd``, then this detection can have multiple
-    true positive predictions matched to it.
+class COCOEvaluationConfig(DetectionEvaluationConfig):
+    """COCO-style evaluation config.
 
-    Dictionaries are added to each predicted/ground truth
-    :class:`fiftyone.core.labels.Detections` instance in the fields listed
-    below; these fields tabulate the true positive (TP), false positive (FP),
-    and false negative (FN) counts for the sample at the specified IoU::
-
-        Ground truth:   detections.<pred_field>_eval
-        Predictions:    detections.<gt_field>_eval
-
-    Dictionaries are also added to each individual
-    :class:`fiftyone.core.labels.Detection` instance in the fields listed
-    below; these fields tabulate the IDs of the matching ground
-    truth/prediction for the detection at the specified IoU::
+    Args:
+        iou (0.75): the IoU threshold to use to determine matches
+        classwise (False): whether to only match objects with the same class
```

Comment:
@ehofesmann I'm curious to get your thoughts on whether `classwise` should be True or False by default. Does COCO-style evaluation use `classwise=True` by default?

It's useful to be able to analyze classification errors for detectors, but with `classwise=False` then matching objects aren't always "TP", which does mean that our TP/FP/FN indicators have an asterisk.

Reply:
I think that it should be `True` by default since that let's you evaluate the model as a whole and is what pycocotools does. Though I added `classwise` because setting it to `False` is really useful to evaluate just the detections. 

This definitely seems like one point that would need to be addressed in our "proper evaluation and metric" initiative we have with Jordi since you should incorporate the detection and classification abilities of a model into a metric separately.

Reply:
ok thanks I reverted to `True` because that's what COCO uses